common.py

```diff
@@ -45,23 +55,72 @@ def nested_process_search(self, criteria, cb_conn, base_query):
 
         elif self.product == "cbth":
             return cbth.nested_process_search(cb_conn, criteria, base_query)
+        
+        elif self.product == "defender": 
+            return defender.nested_process_search(cb_conn, criteria, base_query)
 
     # write the rows of the CSV
     def write_csv(self, output, results, *args):
         for r in results:
             row = [r[0], r[1], r[2], r[3], args[0], args[1]]
             output.writerow(row)
 
-    def get_cbapi_connection(self):
+    def get_connection(self):
         if self.product == 'cbr':
             if self.profile:
                 cb_conn = CbEnterpriseResponseAPI(profile=self.profile)
             else:
                 cb_conn = CbEnterpriseResponseAPI()
 
+            return cb_conn
+
         elif self.product == 'cbth':
             if self.profile:
                 cb_conn = CbThreatHunterAPI(profile=self.profile)
             else:
                 cb_conn = CbThreatHunterAPI()
-        return cb_conn
\ No newline at end of file
+            
+            return cb_conn
+
+    def get_connection_creds(self, creds):
+        if self.profile: 
+            atp_profile = self.profile
+        else: 
+            atp_profile = "default"
+            #atp_profile = "demo"
+        
+        if self.product == 'defender':
+            config = self.config_reader(creds)
+            conn = self.get_atp_aadToken(config[atp_profile]['tenantId'], config[atp_profile]['appId'], config[atp_profile]['appSecret'])
+
+            return conn
+    
+    def get_atp_aadToken(self, tenantID, appID, appSecret):
```

Comment:
```suggestion
    def get_aad_token(self, tenantID, appID, appSecret):
```

Suggest this rename to avoid being tied to the ATP naming scheme and for consistency in casing.